### PR TITLE
By default, smooth out state path by removing adapter 'islands'

### PR DESCRIPTION
**Pull request by kvg**
Adds a (fairly brute-force) method of removing adapter islands by just checking a radius of states around adapters and removing them if they're flanked by the "random" model.

### CODE REVIEW

File: src/annmas/utils/model.py [hunk 1]
```diff
@@ -190,14 +191,57 @@ def build_default_model():
...
+            p_adapter_right = p + 1
+            while p_adapter_right < len(path) and path[p_adapter_right] == path[p]:
+                p_adapter_right = p_adapter_right + 1
+
+            # Now check if this is an adapter island (the surrounding states are just the random model)
+            prev_label = None
+            if p_adapter_left - radius >= 0:
+                prev_labels = set(path[(p_adapter_left - 1 - radius):(p_adapter_left - 1)])
+                if len(prev_labels) == 1:
+                    prev_label = next(iter(prev_labels))
```
Comment (jonn-smith) [inline src/annmas/utils/model.py:214]: This is a little dangerous - sets don't have a guaranteed order.  I think `OrderedSet` would fix this.

File: src/annmas/utils/model.py [hunk 2]
```diff
@@ -190,14 +191,57 @@ def build_default_model():
...
+
+            next_label = None
+            if p_adapter_right + radius < len(path):
+                next_labels = set(path[(p_adapter_right + 1):(p_adapter_right + 1 + radius)])
+                if len(next_labels) == 1:
+                    next_label = next(iter(next_labels))
```
Comment (jonn-smith) [inline src/annmas/utils/model.py:220]: Same thing here.

**Review by jonn-smith — CHANGES_REQUESTED** [review]
Two relatively minor comments.

File: src/annmas/utils/model.py (hunk 1 above)
Comment (kvg) [reply src/annmas/utils/model.py:214]: I'm not sure it matters - the goal is just to check if the previous (next) several states are the exact same value (hence the `len(prev_labels) == 1`, and that that value is 'random'.

File: src/annmas/utils/model.py (hunk 2 above)
Comment (kvg) [reply src/annmas/utils/model.py:220]: See above.

File: src/annmas/utils/model.py (hunk 1 above)
Comment (jonn-smith) [reply src/annmas/utils/model.py:214]: Ah.  Now I understand.  Yeah - this looks good.

**Review by jonn-smith — APPROVED** [review]

**Pull request merged**